pcap.pyx

```diff
@@ -322,20 +325,21 @@ cdef class pcap:
         *args    -- optional arguments passed to callback on execution
         """
         cdef pcap_pkthdr *hdr
-        cdef char *pkt
+        cdef u_char *pkt
         cdef int n
-        cdef int i
-        i = 1
+        cdef int i = 1
         pcap_ex_setup(self.__pcap)
         while 1:
-            Py_BEGIN_ALLOW_THREADS
-            n = pcap_ex_next(self.__pcap, &hdr, &pkt)
-            Py_END_ALLOW_THREADS
+            with nogil:
+                n = pcap_ex_next(self.__pcap, &hdr, &pkt)
             if n == 1:
-                callback(hdr.ts.tv_sec + (hdr.ts.tv_usec / 1000000.0),
-                         PyBuffer_FromMemory(pkt, hdr.caplen), *args)
+                callback(
+                    hdr.ts.tv_sec + (hdr.ts.tv_usec / 1000000.0),
+                    get_buffer(pkt, hdr.caplen),
+                    *args
+                )
             elif n == 0:
-                break
+                continue
```

Comment:
I see this changes the behavior with how timeouts are handled. TODO: mention this in the changelog.